base/applications/clipbrd/clipbrd.c

```diff
@@ -4,15 +4,31 @@
  * PURPOSE:     Provides a view of the contents of the ReactOS clipboard.
  * COPYRIGHT:   Copyright 2015-2018 Ricardo Hanke
  *              Copyright 2015-2018 Hermes Belusca-Maito
+ *              Copyright 2023 Katayama Hirofumi MZ <katayama.hirofumi.mz@gmail.com>
  */
 
 #include "precomp.h"
 
+#include <shlobj.h> /* For CFSTR_... */
+
 static const WCHAR szClassName[] = L"ClipBookWClass";
 
 CLIPBOARD_GLOBALS Globals;
 SCROLLSTATE Scrollstate;
 
+static void InitGlobals(HINSTANCE hInstance)
+{
+    ZeroMemory(&Globals, sizeof(Globals));
+    Globals.hInstance = hInstance;
+
+    /* Registered clipboard formats */
+    Globals.uCFSTR_FILENAMEA = RegisterClipboardFormatA(CFSTR_FILENAMEA);
+    Globals.uCFSTR_FILENAMEW = RegisterClipboardFormatW(CFSTR_FILENAMEW);
+    Globals.uCF_HTML = RegisterClipboardFormatW(L"HTML Format");
+    Globals.uCFSTR_INETURLA = RegisterClipboardFormatA(CFSTR_INETURLA);
+    Globals.uCFSTR_INETURLW = RegisterClipboardFormatW(CFSTR_INETURLW);
```

Comment:
Why are these three last formats necessary here? Usually these should also be accompanied by a readable text format.

Reply:
`CF_HTML` has header and tag info.
The data of `CFSTR_INETURLA/W` can be different from the original text.

Reply:
> Copying an HTML with hyperlink:

That's what we should see.